static/app/views/alerts/wizard/options.tsx

```diff
@@ -187,6 +196,57 @@ export const hideParameterSelectorSet = new Set<AlertType>([
   'cls',
 ]);
 
+const TRANSACTION_SUPPORTED_TAGS = [
+  FieldKey.RELEASE,
+  FieldKey.TRANSACTION,
+  FieldKey.TRANSACTION_OP,
+  FieldKey.TRANSACTION_STATUS,
+  FieldKey.HTTP_METHOD,
+];
+const SESSION_SUPPORTED_TAGS = [FieldKey.RELEASE];
+
+// Some data sets support a very limited number of tags. For these cases,
+// define all supported tags explicitly
+export const DATASET_SUPPORTED_TAGS: Record<Dataset, TagCollection | undefined> =
+  mapValues(
+    {
+      [Dataset.ERRORS]: undefined,
+      [Dataset.TRANSACTIONS]: TRANSACTION_SUPPORTED_TAGS,
+      [Dataset.METRICS]: SESSION_SUPPORTED_TAGS,
+      [Dataset.GENERIC_METRICS]: TRANSACTION_SUPPORTED_TAGS,
+      [Dataset.SESSIONS]: SESSION_SUPPORTED_TAGS,
+    },
+    value => {
+      return value ? makeTagCollection(value) : undefined;
+    }
+  );
+
+// Some data sets support all tags except some. For these cases, define the
+// omissions only
+export const DATASET_OMITTED_TAGS: Record<
+  Dataset,
+  Array<FieldKey | WebVital | MobileVital | SpanOpBreakdown> | undefined
+> = {
+  [Dataset.ERRORS]: [
+    FieldKey.EVENT_TYPE,
+    FieldKey.RELEASE_VERSION,
+    FieldKey.RELEASE_STAGE,
+    FieldKey.RELEASE_BUILD,
+    FieldKey.PROJECT,
+    ...Object.values(WebVital),
+    ...Object.values(MobileVital),
+    ...Object.values(SpanOpBreakdown),
+    FieldKey.TRANSACTION,
+    FieldKey.TRANSACTION_DURATION,
+    FieldKey.TRANSACTION_OP,
+    FieldKey.TRANSACTION_STATUS,
+  ],
+  [Dataset.TRANSACTIONS]: undefined,
+  [Dataset.METRICS]: undefined,
+  [Dataset.GENERIC_METRICS]: undefined,
+  [Dataset.SESSIONS]: undefined,
```

Comment:
whats undefined do here

Reply:
@scttcper  I wanted to be ✨ very explicit ✨ , and using `Record<Dataset, ` requires _every_ dataset to be in the object. Maybe a bit much?